ruby/ql/lib/codeql/ruby/dataflow/internal/SsaImpl.qll

```diff
@@ -473,20 +473,12 @@ class ParameterExt extends TParameterExt {
 private module DataFlowIntegrationInput implements Impl::DataFlowIntegrationInputSig {
   private import codeql.ruby.controlflow.internal.Guards as Guards
 
-  class Parameter = ParameterExt;
-
   class Expr extends Cfg::CfgNodes::ExprCfgNode {
     predicate hasCfgNode(SsaInput::BasicBlock bb, int i) { this = bb.getNode(i) }
   }
 
   Expr getARead(Definition def) { result = Cached::getARead(def) }
 
-  predicate ssaDefAssigns(WriteDefinition def, Expr value) {
-    def.(Ssa::WriteDefinition).assigns(value)
-  }
-
-  predicate ssaDefInitializesParam(WriteDefinition def, Parameter p) { p.isInitializedBy(def) }
-
```

Comment:
Should `ssaDefHasSource` be implemented, restricting to `any(Parameter p).isInitializedBy(def) or def.(Ssa::WriteDefinition).assigns(_)`?

Reply:
Optionally, yes. It makes no semantic difference, it only causes a slight reduction in generated nodes (nodes that are of course unreachable by any step). I figured that such a restriction was likely most of the write definitions, which is why I didn't bother originally, but I just checked, and there is a noticeable count difference, so I'll add it.